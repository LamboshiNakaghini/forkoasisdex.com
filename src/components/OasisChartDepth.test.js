/* global describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import {
  OasisChartDepth,
  mapStateToProps,
  mapDispatchToProps,
} from './OasisChartDepth';

describe('(Component) OasisChartDepth', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps,
    tradingPair: {
      baseToken: 'empty1',
      quoteToken: 'empty2',
    },
  };

  it('should render', () => {
    const wrapper = shallow(
      <OasisChartDepth {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
