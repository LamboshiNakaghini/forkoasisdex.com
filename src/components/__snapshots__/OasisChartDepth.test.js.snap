// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisChartDepth should render 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(38, 166, 154, 0.2)",
          "borderColor": "rgba(38, 166, 154, 1)",
          "borderWidth": 3,
          "data": Array [],
          "hoverBackgroundColor": "#1ABC9C",
          "hoverBorderColor": "#1ABC9C",
          "hoverBorderWidth": 5,
          "invertedStep": true,
          "label": "Buy",
          "pointBackgroundColor": "#1ABC9C",
          "pointBorderColor": "#1ABC9C",
          "pointBorderWidth": 1,
          "pointRadius": 3,
          "pointStyle": "circle",
          "steppedLine": true,
        },
        Object {
          "backgroundColor": "rgba(239, 83, 80, 0.2)",
          "borderColor": "#EF5350",
          "borderWidth": 3,
          "data": Array [],
          "hoverBackgroundColor": "#EF5350",
          "hoverBorderColor": "#EF5350",
          "hoverBorderWidth": 5,
          "label": "Sell",
          "pointBackgroundColor": "#EF5350",
          "pointBorderColor": "#EF5350",
          "pointBorderWidth": 1,
          "pointRadius": 3,
          "pointStyle": "circle",
          "steppedLine": true,
        },
      ],
      "labels": Array [],
    }
  }
  height={180}
  options={
    Object {
      "layout": Object {
        "padding": 5,
      },
      "legend": Object {
        "display": false,
      },
      "scales": Object {
        "xAxes": Array [
          Object {
            "display": true,
            "ticks": Object {
              "callback": [Function],
            },
          },
        ],
        "yAxes": Array [
          Object {
            "scaleLabel": Object {
              "display": false,
              "labelString": "SUM(empty2)",
            },
            "ticks": Object {
              "beginAtZero": true,
            },
          },
        ],
      },
      "tooltips": Object {
        "custom": [Function],
        "enabled": false,
        "mode": "index",
        "position": "nearest",
      },
    }
  }
/>
`;
